File: packages/pdf-generator/android/src/main/java/io/capawesome/capacitorjs/plugins/pdfgenerator/classes/options/GenerateFromHtmlOptions.java
package io.capawesome.capacitorjs.plugins.pdfgenerator.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.pdfgenerator.classes.CustomExceptions;

public class GenerateFromHtmlOptions extends GenerateOptions {

    @Nullable
    private final String baseUrl;

    @NonNull
    private final String html;

    public GenerateFromHtmlOptions(@NonNull PluginCall call) throws Exception {
        super(call);
        this.baseUrl = GenerateFromHtmlOptions.getBaseUrlFromCall(call);
        this.html = GenerateFromHtmlOptions.getHtmlFromCall(call);
    }

    @Nullable
    public String getBaseUrl() {
        return baseUrl;
    }

    @NonNull
    public String getHtml() {
        return html;
    }

    @Nullable
    private static String getBaseUrlFromCall(@NonNull PluginCall call) {
        return call.getString("baseUrl");
    }

    @NonNull
    private static String getHtmlFromCall(@NonNull PluginCall call) throws Exception {
        String html = call.getString("html");
        if (html == null) {
            throw CustomExceptions.HTML_MISSING;
        }
        return html;
    }
}

File: packages/pdf-generator/ios/Plugin/Classes/Options/GenerateFromHtmlOptions.swift
import Foundation
import Capacitor

@objc public class GenerateFromHtmlOptions: GenerateOptions {
    let baseUrl: URL?
    let html: String

    override init(_ call: CAPPluginCall) throws {
        self.baseUrl = GenerateFromHtmlOptions.getBaseUrlFromCall(call)
        self.html = try GenerateFromHtmlOptions.getHtmlFromCall(call)
        try super.init(call)
    }

    private static func getBaseUrlFromCall(_ call: CAPPluginCall) -> URL? {
        guard let baseUrl = call.getString("baseUrl") else {
            return nil
        }
        return URL(string: baseUrl)
    }

    private static func getHtmlFromCall(_ call: CAPPluginCall) throws -> String {
        guard let html = call.getString("html") else {
            throw CustomError.htmlMissing
        }
        return html
    }
}
